server/events/group.js

```diff
@@ -217,11 +217,21 @@ handleGroupOld.flush = function flushGroup() {
 
 function handleGroupNew(event, { hull }) {
   if (event && event.groupId) {
-    const scopedClient = event.userId
-      ? hull
+    const accountIdentity = { external_id: event.groupId };
+
+    const domain = get(event, "traits.domain");
+    if (!isEmpty(domain)) {
```

Comment:
@timliuhull We also should probably validate the data passed in there and see if it's a valid domain ?